src/spaceObjects/playerSpaceship.cpp

```diff
@@ -1633,14 +1635,24 @@ void PlayerSpaceship::onReceiveClientCommand(int32_t client_id, sf::Packet& pack
     case CMD_CUSTOM_FUNCTION:
         {
             string name;
+            int32_t target_id;
             packet >> name;
+            packet >> target_id;
+
             for(CustomShipFunction& csf : custom_functions)
             {
                 if (csf.name == name)
                 {
                     if (csf.type == CustomShipFunction::Type::Button || csf.type == CustomShipFunction::Type::Message)
                     {
-                        csf.callback.call();
+                        P<SpaceObject> target = nullptr;
```

Comment:
I think the code can be simpler, as `getObjectById` will return nullptr on a number that isn't "valid" (including 0 and -1) and a nullptr will translate into a nil on lua.

So it could be reduced to a single line `csf.callback.call(P<PlayerSpaceship>(this), game_server->getObjectById(target_id));`

Reply:
I have already tried this, but `nullptr` in call arguments was not handled yet.

But since you asked... :smile: [Here is the pull request](https://github.com/daid/SeriousProton/pull/55) in Serious Proton.